cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/SparkUtils.java

```diff
@@ -373,35 +351,54 @@ public MethodVisitor visitMethod(final int access, final String name,
           return mv;
         }
 
-        // Generate the method body to just return the hardcode LOCALIZED_CONF_DIR_ZIP file name
-        if (Type.getReturnType(desc).equals(Type.getType(File.class))) {
-          // Spark 1.5+ return type is File
-          // This is what gets generated
-          // return new File(LOCALIZED_CONF_DIR_ZIP);
-          GeneratorAdapter mg = new GeneratorAdapter(mv, access, name, desc);
-          mg.newInstance(Type.getType(File.class));
-          mg.dup();
-          mg.visitLdcInsn(LOCALIZED_CONF_DIR_ZIP);
-          mg.invokeConstructor(Type.getType(File.class), Methods.getMethod(void.class, "<init>", String.class));
+        // Check if it's a recognizable return type.
+        // Spark 1.5+ return type is File
+        boolean isReturnFile = Type.getReturnType(desc).equals(Type.getType(File.class));
+        if (!isReturnFile) {
+          // Spark 1.4 return type is Option<File>
+          if (!Type.getReturnType(desc).equals(Type.getType(Option.class))) {
+            // Unknown type. Not going to modify the code.
+            return mv;
+          }
+          isReturnFile = false;
+        }
+
+        // Generate this first,
+        // SparkConfUtils.createSparkConfZip(this.sparkConf.getAll(), SPARK_CONF_FILE,
+        //                                   LOCALIZED_CONF_DIR, LOCALIZED_CONF_DIR_ZIP);
+        Type sparkConfType = Type.getObjectType("org/apache/spark/SparkConf");
+        GeneratorAdapter mg = new GeneratorAdapter(mv, access, name, desc);
```

Comment:
Those are pretty common in the asm library. Also, all of them are local, so shouldn't be too big of a issue